SimpleWAWS/Models/ResourceGroup.cs

```diff
@@ -174,7 +179,48 @@ public UIResource UIResource
                     AppService = AppService,
                     TemplateName = templateName,
                     IsExtended = IsExtended,
-                    TimeLeftInSeconds = (int)TimeLeft.TotalSeconds
+                    TimeLeftInSeconds = (int)TimeLeft.TotalSeconds,
+                    CsmId= csmId
+                };
+            }
+        }
+
+        public UIResource FunctionsUIResource
```

Comment:
Will investigate . I didnt know what impact that would have on the non-Functions flow.